packages/wrangler/src/experimental-assets.ts

```diff
@@ -1,8 +1,299 @@
+import assert from "node:assert";
 import { existsSync } from "node:fs";
-import path from "node:path";
-import { UserError } from "./errors";
+import { readdir, readFile, stat } from "node:fs/promises";
+import * as path from "node:path";
+import chalk from "chalk";
+import { getType } from "mime";
+import PQueue from "p-queue";
+import prettyBytes from "pretty-bytes";
+import { fetchResult } from "./cfetch";
+import { formatTime } from "./deploy/deploy";
+import { FatalError, UserError } from "./errors";
+import { logger, LOGGER_LEVELS } from "./logger";
+import {
+	BULK_UPLOAD_CONCURRENCY,
+	MAX_ASSET_COUNT,
+	MAX_ASSET_SIZE,
+	MAX_UPLOAD_ATTEMPTS,
+	MAX_UPLOAD_GATEWAY_ERRORS,
+} from "./pages/constants";
+import { hashFile } from "./pages/hash";
+import { APIError } from "./parse";
 import type { Config } from "./config";
 
+type AssetManifest = { [path: string]: { hash: string; size: number } };
+type InitializeAssetsResponse = {
+	// string of file hashes per bucket
+	buckets: string[][];
+	jwt: string;
+};
+
+export type UploadPayloadFile = {
+	name: string;
+	hash: string;
+	contents: string;
+	metadata: {
+		contentType: string;
+	};
+};
+
+type UploadResponse = {
+	jwt?: string;
+};
+
+export const syncAssets = async (
```

Comment:
To keep the two distinct til we land it as stable:

```suggestion
export const syncExperimentalAssets = async (
```